polkadot-parachains/integritee-kusama/src/lib.rs

```diff
@@ -777,6 +787,130 @@ pub type EnsureRootOrAllTechnicalCommittee = EitherOfDiverse<
 	pallet_collective::EnsureProportionAtLeast<AccountId, TechnicalCommitteeInstance, 1, 1>,
 >;
 
+use integritee_parachains_common::porteer::{
+	asset_hub_polkadot_location, forward_teer, ik_cousin_v5, ik_sibling_v5,
+	integritee_runtime_porteer_mint, ip_on_ahp_v5, local_integritee_xcm, IK_FEE,
+};
+use sp_core::hex2array;
+use xcm::{
+	latest::{Location, NetworkId, Parent, SendError},
+	prelude::Parachain,
+};
+use xcm_builder::AliasesIntoAccountId32;
+use xcm_runtime_apis::fees::runtime_decl_for_xcm_payment_api::XcmPaymentApi;
+
+ord_parameter_types! {
+	pub const Alice: AccountId = AccountId::new(hex2array!("d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d"));
+	pub const Ferdie: AccountId = AccountId::new(hex2array!("1cbd2d43530a44705ad088af313e18f80b53ef16b36177cd4b77b846f2a5f07c"));
+}
+
+pub struct PortTokensToPolkadot;
+
+impl PortTokens for PortTokensToPolkadot {
+	type AccountId = AccountId;
+	type Balance = Balance;
+	type Location = Location;
+	type Error = SendError;
+
+	// Todo: Passed owned account id
+	fn port_tokens(
+		who: &Self::AccountId,
+		amount: Self::Balance,
+		location: Option<Self::Location>,
+	) -> Result<(), Self::Error> {
+		let fees = Porteer::xcm_fee_config();
+
+		let xcm1 = local_integritee_xcm(
+			integritee_runtime_porteer_mint(who.clone(), amount, location.clone()),
+			IK_FEE,
+			ik_sibling_v5(),
+			ik_cousin_v5(),
+			((Parent, Parachain(1000)).into(), fees.hop1),
+			(asset_hub_polkadot_location(), fees.hop2),
+			(ip_on_ahp_v5(), fees.hop3),
+		);
+
+		// need to xcms as querying the weight coerces the type to `Xcm<()>`.
+		let xcm2 = local_integritee_xcm(
+			integritee_runtime_porteer_mint(who.clone(), amount, location),
+			IK_FEE,
+			ik_sibling_v5(),
+			ik_cousin_v5(),
+			((Parent, Parachain(1000)).into(), fees.hop1),
+			(asset_hub_polkadot_location(), fees.hop2),
+			(ip_on_ahp_v5(), fees.hop3),
+		);
+
+		let weight = Runtime::query_xcm_weight(VersionedXcm::from(xcm1.clone())).unwrap();
+
+		// Todo: Is this best practice, we could also do this with, but then we have to manually
+		// do some xcm steps on our local chain, e.g. burning assets and adding them to holding:
+		//
+		// let (ticket, delivery_fees) = xcm_config::XcmRouter::validate(
+		// &mut Some((Parent, Parachain(1000)).into()), &mut Some(xcm1),
+		// )?;
+		//
+		// xcm_config::XcmRouter::deliver(ticket)?;
+		PolkadotXcm::execute(RawOrigin::Root.into(), Box::new(VersionedXcm::from(xcm2)), weight)
+			.unwrap();
```

Comment:
Similar case as above with the `forward_teer` function: if we use the `InitiateTransfer` xcm we do not have to do anything manually locally, but we also lose the fine-grained control.

We would ideally want the `who` to pay for the local xcm execution fees instead of root. I guess we need to crawl into the xcm code for this unfortunately too.

Reply:
I think this makes sense.